ballerina/tests/from_json_string_test.bal

```diff
@@ -60,6 +60,17 @@ isolated function testSimpleJsonStringToRecord() returns Error? {
     test:assertEquals(recC.get("b"), 1);
 }
 
+type ReadOnlyUser readonly & record {|
```

Comment:
May this support the scenario something like below,
```ballerina
public type User record {|
    Id id;
    int age;
|};
public type Id readonly & int;
```

Reply:
Yes, added a test